pkg/metrics/server/server.go

```diff
@@ -249,7 +252,11 @@ func (s *defaultServer) Start(ctx context.Context) error {
 
 func (s *defaultServer) createListener(ctx context.Context, log logr.Logger) (net.Listener, error) {
 	if !s.options.SecureServing {
-		return net.Listen("tcp", s.options.BindAddress)
+		if s.options.ListenConfig == nil {
```

Comment:
This should be defaulted, and we should always call it; instead of using the if with net.Listen which is basically just an empty ListenConfig https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/net/dial.go;l=804

Reply:
That's what I had originally but @troy0820  had concerns.

Reply:
My concern was that this default thing @vincepri is asking you to do would have been nil if not set.  So not passing it in the options would have resulted in a nil listenconfig.  

What @vincepri is asking you to do is if this is not set, is to make the options.ListenConfig != nil (creating a default listenconfig) so the listenconfig will be available if not passed through the options.

Reply:
options.ListenConfig cannot be nil is my understanding because its by value

Reply:
I ran this just to prove my theory
```
package main

import (
	"fmt"
	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
)

func main() {
	opt := server.Options{}
	fmt.Println(opt.ListenConfig.KeepAlive)
}
```

running that prints
```
0s
```

Reply:
You are correct.  This works because it's not a pointer, so it's nil value is a listenConfig. You never made it a pointer, and this will default to a correct listenConfig.  We can resolve this.